src/main/java/seedu/address/logic/commands/DeleteRestaurantNoteCommand.java

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Deletes a note of an existing restaurant in the restaurant book.
+ */
+public class DeleteRestaurantNoteCommand extends Command {
+    public static final String COMMAND_WORD = "(rt)deletenote";
+    public static final String COMMAND_FUNCTION = "Deletes the food note of the restaurant identified "
+            + "by the index number used in the last restaurant listing.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": " + COMMAND_FUNCTION + "\n"
+            + "Parameters: INDEX and LINE_NUMBER (must be a positive integer) "
+            + "[" + PREFIX_LINE_NUMBER_RECOMMENDED + "LINE_NUMBER] "
+            + "[" + PREFIX_LINE_NUMBER_GOOD + "LINE_NUMBER] "
+            + "[" + PREFIX_LINE_NUMBER_BAD + "LINE_NUMBER]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_LINE_NUMBER_RECOMMENDED + "2 "
+            + PREFIX_LINE_NUMBER_GOOD + "1 "
+            + PREFIX_LINE_NUMBER_BAD + "3";
+    public static final String MESSAGE_DELETE_NOTE_SUCCESS = "Deleted notes for restaurant: %1$s";
+
+    private final Index index;
+    private final ArrayList<Integer> lineRec;
+    private final ArrayList<Integer> lineGood;
+    private final ArrayList<Integer> lineBad;
+
+    /**
+     * @param index of the restaurant in the filtered restaurant list to edit the note
+     * @param lineRec number of a specific note in the recommended food information stored
+     * @param lineGood number of a specific note in the good food information stored
+     * @param lineBad number of a specific note in the bad food information stored
+     */
+    public DeleteRestaurantNoteCommand(Index index, ArrayList<Integer> lineRec, ArrayList<Integer> lineGood,
+                                       ArrayList<Integer> lineBad) {
+        requireAllNonNull(index, lineRec, lineGood, lineBad);
+
+        this.index = index;
+        this.lineRec = lineRec;
+        this.lineGood = lineGood;
+        this.lineBad = lineBad;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Restaurant> lastShownList = model.getFilteredRestaurantList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_RESTAURANT_DISPLAYED_INDEX);
+        }
+
+        Restaurant restaurantToEdit = lastShownList.get(index.getZeroBased());
+        ArrayList<Note> recNotes = new ArrayList<>();
+        ArrayList<Note> goodNotes = new ArrayList<>();
+        ArrayList<Note> badNotes = new ArrayList<>();
+
+        if (lineRec.size() != 0) {
+            if (restaurantToEdit.getRecommendedFood().size() == 0) {
+                throw new CommandException(Messages.MESSAGE_NO_RECOMMENDED_FOOD);
+            }
+            for (Integer i : lineRec) {
+                if (i > restaurantToEdit.getRecommendedFood().size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_LINE_NUMBER_REC);
```

Comment:
Not sure whether you want to throw exception or just carry on with the remaining valid lines O: